commit 44b70d1fd3e5be15a7853f54b92235a65fec6679
Author: Philipp von Weitershausen <philikon@silklabs.com>
Date:   Wed Sep 7 15:17:40 2016 -0700

    react-native fix: allow rn-cli.config.js to specify the default transformer, again
    
    See https://github.com/facebook/react-native/pull/9799

diff --git a/node_modules/react-native/local-cli/bundle/buildBundle.js b/node_modules/react-native/local-cli/bundle/buildBundle.js
index 337340d..7a4eb8e 100644
--- a/node_modules/react-native/local-cli/bundle/buildBundle.js
+++ b/node_modules/react-native/local-cli/bundle/buildBundle.js
@@ -40,13 +40,18 @@ function buildBundle(args, config, output = outputBundle, packagerInstance) {
   if (!packagerInstance) {
     let assetExts = (config.getAssetExts && config.getAssetExts()) || [];
 
+    const transformModulePath =
+      args.transformer ? path.resolve(args.transformer) :
+      typeof config.getTransformModulePath === 'function' ? config.getTransformModulePath() :
+      undefined;
+
     const options = {
       projectRoots: config.getProjectRoots(),
       assetExts: defaultAssetExts.concat(assetExts),
       assetRoots: config.getAssetRoots(),
       blacklistRE: config.getBlacklistRE(args.platform),
       getTransformOptionsModulePath: config.getTransformOptionsModulePath,
-      transformModulePath: args.transformer,
+      transformModulePath: transformModulePath,
       extraNodeModules: config.extraNodeModules,
       nonPersistent: true,
       resetCache: args.resetCache,
diff --git a/node_modules/react-native/local-cli/bundle/bundleCommandLineArgs.js b/node_modules/react-native/local-cli/bundle/bundleCommandLineArgs.js
index 46bb727..41a3478 100644
--- a/node_modules/react-native/local-cli/bundle/bundleCommandLineArgs.js
+++ b/node_modules/react-native/local-cli/bundle/bundleCommandLineArgs.js
@@ -18,8 +18,7 @@ module.exports = [
     default: 'ios',
   }, {
     command: '--transformer [string]',
-    description: 'Specify a custom transformer to be used (absolute path)',
-    default: require.resolve('../../packager/transformer'),
+    description: 'Specify a custom transformer to be used',
   }, {
     command: '--dev [boolean]',
     description: 'If false, warnings are disabled and the bundle is minified',
diff --git a/node_modules/react-native/local-cli/dependencies/dependencies.js b/node_modules/react-native/local-cli/dependencies/dependencies.js
index 633a594..7dfecee 100644
--- a/node_modules/react-native/local-cli/dependencies/dependencies.js
+++ b/node_modules/react-native/local-cli/dependencies/dependencies.js
@@ -89,7 +89,6 @@ module.exports = {
       description: 'The platform extension used for selecting modules',
     }, {
       command: '--transformer [path]',
-      default: null,
       description: 'Specify a custom transformer to be used'
     }, {
       command: '--verbose',
diff --git a/node_modules/react-native/local-cli/server/server.js b/node_modules/react-native/local-cli/server/server.js
index 65bd4ad..6ed2191 100644
--- a/node_modules/react-native/local-cli/server/server.js
+++ b/node_modules/react-native/local-cli/server/server.js
@@ -109,8 +109,7 @@ module.exports = {
     description: 'Disable file watcher'
   }, {
     command: '--transformer [string]',
-    default: require.resolve('../../packager/transformer'),
-    description: 'Specify a custom transformer to be used (absolute path)'
+    description: 'Specify a custom transformer to be used'
   }, {
     command: '--reset-cache, --resetCache',
     description: 'Removes cached files',
